build/Microsoft.DotNet.Cli.Test.targets

```diff
@@ -39,19 +39,21 @@
       <PathSeparator>/</PathSeparator>
       <PathSeparator Condition=" '$(OSName)' == 'win' ">\</PathSeparator>
       <TestPackageBuildVersionSuffix>$(CommitCount)</TestPackageBuildVersionSuffix>
-      <TestOutputDir>$(BaseOutputDirectory)/tests</TestOutputDir>
-      <TestPackagesBuildDir>$(TestOutputDir)/packagesBuild</TestPackagesBuildDir>
-      <TestPackagesDir>$(TestOutputDir)/packages</TestPackagesDir>
-      <TestArtifactsDir>$(TestOutputDir)/artifacts</TestArtifactsDir>
-      <TestResultXmlDir>$(TestOutputDir)/results</TestResultXmlDir>
-      <TestDirectory>$(RepoRoot)/test</TestDirectory>
-      <DotnetUnderTest>$(DotnetStage2)</DotnetUnderTest>
+      <TestOutputDir>$(BaseOutputDirectory)/tests/</TestOutputDir>
+      <TestPackagesBuildDir>$(TestOutputDir)/packagesBuild/</TestPackagesBuildDir>
+      <TestPackagesDir>$(TestOutputDir)/packages/</TestPackagesDir>
+      <TestArtifactsDir>$(TestOutputDir)/artifacts/</TestArtifactsDir>
+      <TestResultXmlDir>$(TestOutputDir)/results/</TestResultXmlDir>
+      <TestDirectory>$(RepoRoot)/test/</TestDirectory>
+      <DotnetUnderTest>$(Stage2Directory)</DotnetUnderTest>
```

Comment:
What do you think about making this conditional, so it can be overridden?

For example `<DotnetUnderTest Condition=" '$(DotnetUnderTest)' == '' ">$(Stage2Directory)</DotnetUnderTest>`

Then we could easily reuse this for testing other dotnets, in perhaps the installers, or during dev time.